torch/_inductor/codegen/cuda/cutlass_utils.py

```diff
@@ -0,0 +1,170 @@
+
+    if arch == "90":
+        cutlass_generator.GenerateSM90(manifest, args.cuda_version)
+        cutlass_generator.GenerateSM80(manifest, args.cuda_version)
+    else:
+        try:
+            func = getattr(cutlass_generator, "GenerateSM" + arch)
+            func(manifest, args.cuda_version)
+        except AttributeError as e:
+            raise NotImplementedError(
+                "Arch " + arch + " is not supported by current cutlass lib."
+            ) from e
+
+    return manifest.operations
+
+
+def torch_dtype_match(torch_dtype0: torch.dtype, torch_dtype1: torch.dtype) -> bool:
+    _MATCHED_DTYPES = [
+        {torch.float, torch.float32},
+        {torch.float16, torch.half},
+    ]
+
+    if torch_dtype0 == torch_dtype1:
+        return True
+    for matched_dtypes in _MATCHED_DTYPES:
+        if torch_dtype0 in matched_dtypes and torch_dtype1 in matched_dtypes:
+            return True
+    return False
+
+
+def dtype_match(torch_dtype: torch.dtype, cutlass_dtype: cutlass_lib.DataType) -> bool:
+    if torch_dtype == torch.float or torch_dtype == torch.float32:
+        return (
+            cutlass_dtype == cutlass_lib.DataType.f32
+            or cutlass_dtype == cutlass_lib.DataType.tf32
+        )
+    elif torch_dtype == torch.float16 or torch_dtype == torch.half:
+        return cutlass_dtype == cutlass_lib.DataType.f16
+    elif torch_dtype == torch.bfloat16:
+        return cutlass_dtype == cutlass_lib.DataType.bf16
+    else:
+        return False
+
+
+def get_accumulator_dtype(input_torch_dtypes: List[torch.dtype]) -> torch.dtype:
+    """
+    Given a list of input torch dtypes, returns the inferred accumulator torch dtype.
+    """
+
+    if len(input_torch_dtypes) == 0:
+        return None
+    torch_dtype = input_torch_dtypes[0]
+    for dtype in input_torch_dtypes[1:]:
+        if not torch_dtype_match(torch_dtype, dtype):
+            raise RuntimeError(f"Unmatched input dtypes: {torch_dtype=}, {dtype=}")
+    if torch_dtype in {torch.float16, torch.half}:
+        if torch.backends.cuda.matmul.allow_fp16_reduced_precision_reduction:
+            return torch_dtype
+        else:
+            return torch.float
+    if torch_dtype in {torch.bfloat16, torch.float, torch.float32}:
+        return torch.float
+    raise NotImplementedError(f"Unsupported data type: {input_torch_dtypes=}")
+
+
+def get_alignments(torch_dtype: torch.dtype) -> List[int]:
+    """
+    Returns all possible number of elements for a dtype which forms valid cutlass alignments.
+    CUTLASS gemm / conv APIs support 16 bytes max alignment, and 2 bytes min alignment.
+    """
+
+    if torch_dtype in (torch.float16, torch.half, torch.bfloat16):
+        return [8, 4, 2, 1]
+    elif torch_dtype in (torch.float, torch.float32):
+        return [4, 2, 1]
+    else:
+        raise NotImplementedError(f"unsupported {torch_dtype=} for alignments")
+
+
+def get_alignment(inductor_layout: Layout) -> int:
```

Comment:
Nit: rename this to `get_max_alignment`? Because that's what the docstring says.